# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/FontFaceSet.h
# Commit: e3f8526e0e2a
# Full Hash: e3f8526e0e2a08a4a8342dccedf05bebb3de1baf
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-09-08 21:39:05
# Regressor Bug: 1715501
# File Overlap Count: 1
# Description:
#   Bug 1715501 - patch 2 - Pass the presContext to platform font lookup methods so they can query it for font visibility. r=emilio
#   
#   This does not in itself change user-visible behavior, but is a foundation for bug
#   1715507 where we will make the behavior per-context instead of global. This is basically
#   just plumbing, passing a pointer to the presContext that's asking for fonts to be
# ==============================================================================

diff -r cfdd1c5d8d3f -r e3f8526e0e2a layout/style/FontFaceSet.h
--- a/layout/style/FontFaceSet.h	Wed Sep 08 09:35:16 2021 +0000
+++ b/layout/style/FontFaceSet.h	Wed Sep 08 09:35:17 2021 +0000
@@ -66,6 +66,10 @@
                           : nullptr;
     }
 
+    nsPresContext* GetPresContext() const final {
+      return mFontFaceSet ? mFontFaceSet->GetPresContext() : nullptr;
+    }
+
     bool IsFontLoadAllowed(const gfxFontFaceSrc&) final;
 
     void DispatchFontLoadViolations(
